src/llama_stack/providers/remote/vector_io/qdrant/qdrant.py

```diff
@@ -128,7 +128,42 @@ async def query_vector(self, embedding: NDArray, k: int, score_threshold: float)
         return QueryChunksResponse(chunks=chunks, scores=scores)
 
     async def query_keyword(self, query_string: str, k: int, score_threshold: float) -> QueryChunksResponse:
-        raise NotImplementedError("Keyword search is not supported in Qdrant")
+        try:
+            results = (
+                await self.client.query_points(
+                    collection_name=self.collection_name,
+                    query_filter=models.Filter(
+                        must=[
+                            models.FieldCondition(
+                                key="chunk_content.content", match=models.MatchText(text=query_string)
+                            )
+                        ]
+                    ),
+                    limit=k,
+                    with_payload=True,
+                    with_vectors=False,
+                    score_threshold=score_threshold,
+                )
+            ).points
+        except Exception as e:
+            log.error(f"Error querying keyword search in Qdrant collection {self.collection_name}: {e}")
+            raise
+
+        chunks, scores = [], []
+        for point in results:
+            assert isinstance(point, models.ScoredPoint)
+            assert point.payload is not None
```

Comment:
nit: should we replace assertions in production code with explicit checks? It may introduce a couple more lines of code here, but could be a safer option.

See here: https://discuss.python.org/t/python-can-we-use-assert-key-word-in-production-code/94316/4